lib/playload/repositories/base_repository.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module BaseRepository
+  def find_by(conditions)
+    fetch_relation.where(conditions).limit(1).one
+  end
+
+  def find_with(id, includes: [])
+    aggregate(*includes).where(id: id).map_to(self.class.entity).one
+  end
+
+  def find_by_with(conditions, includes: [])
+    aggregate(*includes).where(conditions).map_to(self.class.entity).one
+  end
+
+  def all_by(conditions)
+    relation_by(conditions).to_a
```

Comment:
Out of interest, what is the difference between this and `fetch_relation.where(conditions)` ?

Reply:
Good spot, that's a typo